File: CourierAmexAPI/CourierAmex/Models/AreaModel.cs
namespace CourierAmex.Models
{
    public class AreaModel : BaseEntityModel<int>
    {
        public required int ZoneId { get; set; }
        public required string Code { get; set; }
        public required string Name { get; set; }
        public string? Notes { get; set; }

        public string? ZoneName { get; set; }
        public int? StateId { get; set; }
        public int? CountryId { get; set; }
    }
}

File: CourierAmexClient/src/app/features/general/models/area.model.ts
import { BaseEntity } from "@app/models";

export interface AreaModel extends BaseEntity<number> {
  zoneId?: number;
  code: string;
  name: string;
  notes?: string;
  
  zoneName?: string;
  countryId?: number;
  stateId?: number;
}

export const newAreaModel: AreaModel = {
  id: 0,
  countryId: undefined,
  stateId: undefined,
  zoneId: undefined,
  code: '',
  name: '',
  status: 2
}
